CodePush.js

```diff
@@ -204,93 +178,83 @@ function sync(options = {}, syncStatusChangeCallback, downloadProgressCallback)
-            
-            // Since this is an optional update, add another button
-            // to allow the end-user to ignore it       
-            dialogButtons.push({
-              text: syncOptions.updateDialog.optionalIgnoreButtonLabel,
-              onPress: () => {
-                syncStatusChangeCallback(CodePush.SyncStatus.UPDATE_IGNORED);
-                resolve(CodePush.SyncStatus.UPDATE_IGNORED);
-              }
-            });
+      return await new Promise((resolve, reject) => {  
+        let message = null;
+        let dialogButtons = [{
+          text: null,
+          onPress: async () => { 
+            resolve(await doDownloadAndInstall());
           }
-          
-          // If the update has a description, and the developer
-          // explicitly chose to display it, then set that as the message
-          if (syncOptions.updateDialog.appendReleaseDescription && remotePackage.description) {
-            message += `${syncOptions.updateDialog.descriptionPrefix} ${remotePackage.description}`;  
-          }
-          
-          syncStatusChangeCallback(CodePush.SyncStatus.AWAITING_USER_ACTION);
-          Alert.alert(syncOptions.updateDialog.title, message, dialogButtons);
+        }];
+        
+        if (remotePackage.isMandatory) {
+          message = syncOptions.updateDialog.mandatoryUpdateMessage;
+          dialogButtons[0].text = syncOptions.updateDialog.mandatoryContinueButtonLabel;
         } else {
-          doDownloadAndInstall();
+          message = syncOptions.updateDialog.optionalUpdateMessage;
+          dialogButtons[0].text = syncOptions.updateDialog.optionalInstallButtonLabel;        
+          // Since this is an optional update, add another button
+          // to allow the end-user to ignore it       
+          dialogButtons.push({
+            text: syncOptions.updateDialog.optionalIgnoreButtonLabel,
+            onPress: () => {
+              syncStatusChangeCallback(CodePush.SyncStatus.UPDATE_IGNORED);
+              resolve(CodePush.SyncStatus.UPDATE_IGNORED);
+            }
+          });
         }
-      })
-      .catch(rejectPromise)
-      .done();
-  });     
+        
+        // If the update has a description, and the developer
+        // explicitly chose to display it, then set that as the message
+        if (syncOptions.updateDialog.appendReleaseDescription && remotePackage.description) {
+          message += `${syncOptions.updateDialog.descriptionPrefix} ${remotePackage.description}`;  
+        }
+         
+        syncStatusChangeCallback(CodePush.SyncStatus.AWAITING_USER_ACTION);
+        Alert.alert(syncOptions.updateDialog.title, message, dialogButtons);
+      });
+    } else {
+      return await doDownloadAndInstall();
+    }
+  } catch (error) {
+    syncStatusChangeCallback(CodePush.SyncStatus.UNKNOWN_ERROR);
+    log(error.message); 
+    throw error;
+  } 
 };
 
 var CodePush = {
```

Comment:
There are a few properties whose key and value are the same and can be simplified (e.g. "checkForUpdate: checkForUpdate," -> "checkForUpdate,")